### ISSUE TITLE
Disabled internal pretraining with 0 known_cls_ratio

### ISSUE THREAD

**Issue by kwonmha** [fanolabs/NID_ACLARR2022#3]
I wonder why you disabled internal pretraining when `args.known_cls_ratio == 0`.
It becomes unsupervised setting as you wrote in the paper but unsupervised setting also involves internal pretraining - MTP.

**Comment by kwonmha** [comment]
Please let me know how to reproduce your results on unsupervised settings.

**Comment by kwonmha** [comment]
Well, it seems that supervised learning with external labeled data by cross entropy loss on MPT stage is not implemented here.
Because in `dataloader.Data` class, `self.train_labeled_dataloader` is never assigned after decalared in `__init__()`.
So iterating over `data.train_labeled_dataloader` in `InternalPretrainModelManager.train()` fails.

**Comment by zhang-yu-wei** [comment]
Hi,

Thanks for bringing this up!

As indicated in the paper, internal pretraining is only applied on semi-supervised setting.
<img width="308" alt="Screen Shot 2022-09-12 at 11 43 10" src="https://user-images.githubusercontent.com/33276780/189731698-5fa513f6-6ed4-40d7-81de-d16720ad7060.png">

Let me know if you have any questions.

**Comment by kwonmha** [comment]
Thanks for reply.

I thought "This step" in the captured paper segment you attached means "continual pretraining with the annotated data(D-labeled-known)", not MTP.
Q1. Then, what's the difference between MTP in Table2(performance on unsupervised NID) and internal pretraining?

The reason I'm asking this is because I want to reproduce the results of unsupervised setting.
Q2. Is only semi-supervised setting implemented in this repository?

**Comment by zhang-yu-wei** [comment]
Maybe I misunderstood your question before. Internal pretrain represents the pretraining conducted with annotated data from current domain. It only applies on semi-supervised setting where annotated data is available. External pretrain represents the pretraining conducted on external datasets (e.g. CLINC).

Based on this agreement, I think the answer to your questions should be:
A1. MTP in Table.2 does not apply internal pretrain, but MTP in Table.3 applies internal pretrain.
A2. Both semi-supervised and unsupervised settings are implemented here. To conduct unsupervised training, `args.known_cls_ratio = 0` should be set so that we do not apply internal pretrain, and the entire training is conducted on unlabeled data.

**Comment by kwonmha** [comment]
Thanks for clarifying.

Then, for reproducting MTP in table 2, it seems that external pretrained model should be prepared and trained by CLNN .
I tried to find external pretrained model in the repo you linked and couldn't find model trained with CLINC.
How can I get the external pretrained model trained with CLINC dataset as written in your paper?

Is it OK to use other models like IntentBert-banking, IntentBert-mcid, IntentBert-stackoverflow ?

**Comment by zhang-yu-wei** [comment]
Yes, IntentBert-banking is pretrained with labeled data from CLINC and unlabeled data from banking. So it is unsupervised.

**Comment by kwonmha** [comment]
Thanks you!
I think I got all things understood clearly.
